src/simulation/TxGenerator.cpp

```diff
@@ -722,6 +756,106 @@ TxGenerator::invokeSorobanLoadTransactionV2(
     return std::make_pair(account, tx);
 }
 
+std::pair<TxGenerator::TestAccountPtr, TransactionFrameBaseConstPtr>
+TxGenerator::invokeSACPayment(uint32_t ledgerNum, uint64_t fromAccountId,
+                              SCAddress const& toAddress,
+                              ContractInstance const& instance, uint64_t amount,
+                              std::optional<uint32_t> maxGeneratedFeeRate)
+{
+    auto fromAccount = findAccount(fromAccountId, ledgerNum);
+    fromAccount->loadSequenceNumber();
+
+    SCVal fromVal(SCV_ADDRESS);
+    fromVal.address() = makeAccountAddress(fromAccount->getPublicKey());
+
+    SCVal toVal(SCV_ADDRESS);
+    toVal.address() = toAddress;
+
+    Operation op;
+    op.body.type(INVOKE_HOST_FUNCTION);
+    auto& ihf = op.body.invokeHostFunctionOp().hostFunction;
+    ihf.type(HOST_FUNCTION_TYPE_INVOKE_CONTRACT);
+    ihf.invokeContract().contractAddress = instance.contractID;
+    ihf.invokeContract().functionName = "transfer";
+
+    ihf.invokeContract().args = {fromVal, toVal, makeI128(amount)};
+
+    SorobanResources resources;
+    // These are overestimated
+    resources.writeBytes = 800;
+    resources.diskReadBytes = 800;
+    resources.instructions = 250'000;
+
+    std::vector<uint32_t> archivedEntries;
+    bool autorestore = false;
+    // If the instance has expired, autorestore all keys
+    // The contract invocation will extend the TTL of the contract
+    // and the balance entries, so autorestore is only used in
+    // some edge cases.
+    if (!isLive(instance.readOnlyKeys.back(), ledgerNum))
+    {
+        resources.footprint.readWrite.emplace_back(
+            instance.readOnlyKeys.back());
+        archivedEntries = {0};
+        autorestore = true;
+    }
+    else
+    {
+        resources.footprint.readOnly = instance.readOnlyKeys;
+    }
```

Comment:
I don't think this belongs here. To me, the purpose of SAC load is to test parallelism and benchmark soroban via the "minimal" unit of a useful TX. This autorestore introduces arbitrary IO work that can't be controlled via any of our flags and isn't very predictable. IMO we should just raise the minimum TTL such that we never need to restore SAC entries (which I thought we already did). The non-sac instance generates autorestores already based on the disk IO distribution via pre-populated archived entries.

Reply:
Yeah this was an artifact of my initial testing. It's no longer required. Removed.